src/main/java/uk/gov/companieshouse/web/accounts/service/smallfull/impl/BalanceSheetServiceImpl.java

```diff
@@ -72,66 +83,152 @@ public BalanceSheet getBalanceSheet(String transactionId, String companyAccounts
+        LastAccountsApi lastAccountsApi = companyProfile.getAccounts().getLastAccounts();
+        if (isMultipleYearFiler(companyProfile)) {
+            LocalDate previousPeriodStartOn = convertDateTimeToLocalDate(lastAccountsApi.getPeriodStartOn());
+            LocalDate previousPeriodEndOn = convertDateTimeToLocalDate(lastAccountsApi.getPeriodEndOn());
 
-            LocalDate previousPeriodEndOn =
-                convertDateTimeToLocalDate(companyProfile.getAccounts().getLastAccounts().getPeriodEndOn());
-            isSameYear = accountsDatesHelper.isSameYear(previousPeriodEndOn, currentPeriodEndOn);
+            return accountsDatesHelper.generateBalanceSheetHeading(previousPeriodStartOn, previousPeriodEndOn, isSameYear);
         }
+        return null;
+    }
 
-        String currentPeriodHeading = accountsDatesHelper.generateBalanceSheetHeading(currentPeriodStartOn, currentPeriodEndOn, isSameYear);
+    private boolean isSameYear(CompanyProfileApi companyProfile) {
+        if (isMultipleYearFiler(companyProfile)) {
+            LastAccountsApi lastAccountsApi = companyProfile.getAccounts().getLastAccounts();
+            LocalDate previousPeriodEndOn = convertDateTimeToLocalDate(lastAccountsApi.getPeriodEndOn());
 
-        BalanceSheetHeadings balanceSheetHeadings = new BalanceSheetHeadings();
-        balanceSheetHeadings.setCurrentPeriodHeading(currentPeriodHeading);
+            NextAccountsApi nextAccountsApi = companyProfile.getAccounts().getNextAccounts();
+            LocalDate currentPeriodEndOn = convertDateTimeToLocalDate(nextAccountsApi.getPeriodEndOn());
 
-        return balanceSheetHeadings;
+            return accountsDatesHelper.isSameYear(previousPeriodEndOn, currentPeriodEndOn);
+        }
+        return false;
     }
 
-    private LocalDate convertDateTimeToLocalDate(DateTime dateTime) {
+    private CompanyProfileApi getCompanyProfile(String companyNumber) throws ServiceException {
+        return companyService.getCompanyProfile(companyNumber);
+    }
+
+    private String createUri(UriTemplate uri, String transactionId, String companyAccountsId) {
+        return uri.expand(transactionId, companyAccountsId).toString();
+    }
+
+    private boolean isMultipleYearFiler(CompanyProfileApi companyProfile) {
+        LastAccountsApi lastAccountsApi = companyProfile.getAccounts().getLastAccounts();
+        return lastAccountsApi != null && lastAccountsApi.getPeriodEndOn() != null;
+    }
 
-        Date date = new Date(dateTime.getValue());
-        return accountsDatesHelper.convertDateToLocalDate(date);
+    private boolean isCurrentPeriodCreated(Map<String, String> links) {
+        return isCreated(links, SmallFullLinkType.CURRENT_PERIOD);
+    }
+
+    private boolean isPreviousPeriodCreated(Map<String, String> links) {
+        return isCreated(links, SmallFullLinkType.PREVIOUS_PERIOD);
+    }
+
+    private boolean isCreated(Map<String, String> links, SmallFullLinkType linkType) {
```

Comment:
The reason it was done this way was because I felt instead of having a `isPreviousPeriodCreated` method that excepts 2 parameters which can increase the line length, it may be easier to just have a method that does it all for you.

Regarding the class name, I have followed what is done in the API and I didn't want to change because then we are out of sync with naming. So I feel the same change would have to be made there as well to keep things consistent